drivers/periph_common/init.c

```diff
@@ -73,6 +122,7 @@ void periph_init(void)
     /* initialize configured I2C devices */
 #ifdef MODULE_PERIPH_INIT_I2C
     for (unsigned i = 0; i < I2C_NUMOF; i++) {
+        _i2c_sync_devices(i);
```

Comment:
Why not move this into i2c_init() ?

There may be weird cases where users manually initialize their i2C peripherals without relying on `auto_init`

Reply:
True. I'll check that. It will however require touching the i2c_init files of all cpus. Maybe it makes more sense to include it in this auto_init but also provide it in the form of a function whenever periph_i2c_reconfigure is available so that people can deliberately use it at any time if they want to manually set up things?